### PR TITLE
ORC-301. `extractFileTail` should open a file in `try` statement

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
This PR moves the file open position to prevent opened file leakage more robustly.

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
Hi, @omalley .
Could you review this PR?

File: java/core/src/java/org/apache/orc/impl/ReaderImpl.java [hunk 1]
```diff
@@ -516,12 +516,13 @@ OrcTail buildEmptyTail() {
 
   protected OrcTail extractFileTail(FileSystem fs, Path path,
       long maxFileLength) throws IOException {
-    FSDataInputStream file = fs.open(path);
+    FSDataInputStream file = null;
     ByteBuffer buffer;
     OrcProto.PostScript ps;
     OrcProto.FileTail.Builder fileTailBuilder = OrcProto.FileTail.newBuilder();
     long modificationTime;
     try {
```
Comment (prasanthj) [inline java/core/src/java/org/apache/orc/impl/ReaderImpl.java]: Maybe replace with try-with-resource block?

Comment (dongjoon-hyun) [reply java/core/src/java/org/apache/orc/impl/ReaderImpl.java]: Thank you for review, @prasanthj .
Ya. I thought to use `try-with-resource` first, but it needs more change to print out the log at [Line 610](https://github.com/apache/orc/pull/218/files#diff-90ceebe6c0bfa3e98b4536455a60b498R610).

Comment (dongjoon-hyun) [reply java/core/src/java/org/apache/orc/impl/ReaderImpl.java]: I think this is the minimal change while keeping the current behavior.

Comment (prasanthj) [reply java/core/src/java/org/apache/orc/impl/ReaderImpl.java]: Line 610 doesn't add much value as close() will throw IOException anyway which should be sufficient to know the reason.

Comment (dongjoon-hyun) [reply java/core/src/java/org/apache/orc/impl/ReaderImpl.java]: So, you mean it's okay to remove the following in this PR, right?
```
catch (IOException ex) {
  LOG.error("Failed to close the file after another error", ex);
}
```

Comment (prasanthj) [reply java/core/src/java/org/apache/orc/impl/ReaderImpl.java]: Yeah that's right.

**Comment by dongjoon-hyun** [comment]
Thank you, @prasanthj . I updated according to your advice.

**Comment by prasanthj** [comment]
lgtm, +1

**Comment by dongjoon-hyun** [comment]
Thank you so much, @prasanthj !

**Comment by dongjoon-hyun** [comment]
Thank you for merging, @prasanthj !

**Pull request closed without merging**